### PR TITLE
DOCS-763: Update the out of date Python SDK module code in Extend 

### PR DESCRIPTION
**Pull request by sguequierre**
Part of Michael Lee's feedback from DOCS-763

- Used Python SDK's example module to base these updates off of: https://github.com/viamrobotics/viam-python-sdk/tree/main/examples/simple_module so I believe Cheuk or Nicole are the relevant SMEs! 

This fixed @abe-winter 's module while he was working on it in office so is prob good change

### CODE REVIEW

File: docs/extend/modular-resources/create/_index.md [hunk 1]
```diff
@@ -63,24 +63,26 @@ The Go code for the custom model (<file>mybase.go</file>) and module entry point
 {{< tabs name="Sample SDK Code">}}
 {{% tab name="Python"%}}
 
-<file>my_base.py</file> implements a custom model of the base component built-in resource, "mybase".
+<file>my_base_client.py</file> implements a custom model of the base component built-in resource, "mybase".
```
Comment (cheukt) [inline docs/extend/modular-resources/create/_index.md:66]: I find this line a bit confusing, also my_base_client would be an inaccurate name - my_base_model/my_base would be better.
```suggestion
<file>my_base.py</file> implements "mybase", a custom model of the base component.
```

File: docs/extend/modular-resources/create/_index.md [hunk 2]
```diff
@@ -154,6 +152,14 @@ class MyBase(Base, Reconfigurable):
         self.left.set_power(power=((linear.y - angular.z) / sum), extra=extra, timeout=timeout)
         self.right.set_power(power=((linear.y + angular.z) / sum), extra=extra, timeout=timeout)
 
+    # set_velocity: unimplemented
+    async def set_velocity(self, linear: Vector3, angular: Vector3, *, extra: Optional[Dict[str, Any]] = None, timeout: Optional[float] = None, **kwargs):
+        pass
+
+    # get_properties: unimplemented
```
Comment (sguequierre) [inline docs/extend/modular-resources/create/_index.md:159]: This was added and otherwise module breaks: will add for go in separate PR as more feedback to simplify Python module instructions here!

File: docs/extend/modular-resources/create/_index.md [hunk 3]
```diff
@@ -154,6 +152,14 @@ class MyBase(Base, Reconfigurable):
         self.left.set_power(power=((linear.y - angular.z) / sum), extra=extra, timeout=timeout)
         self.right.set_power(power=((linear.y + angular.z) / sum), extra=extra, timeout=timeout)
 
+    # set_velocity: unimplemented
+    async def set_velocity(self, linear: Vector3, angular: Vector3, *, extra: Optional[Dict[str, Any]] = None, timeout: Optional[float] = None, **kwargs):
+        pass
```
Comment (cheukt) [inline docs/extend/modular-resources/create/_index.md:157]: prefer `raise NotImplementedError`, pass won't error out here when called which is sometimes more confusing

File: docs/extend/modular-resources/create/_index.md [hunk 4]
```diff
@@ -162,35 +168,13 @@ class MyBase(Base, Reconfigurable):
...
-Registry.register_resource_creator(Base.SUBTYPE, MyBase.MODEL, ResourceCreatorRegistration(MyBase.new_base, MyBase.validate_config))
 ```
 
-</details>
-
 {{% /tab %}}
 {{% tab name="Go"%}}
 
-<file>mybase.go</file> implements a custom model of the base component built-in resource, "mybase", and registers the new model and API helper functions with the SDK.
+<file>mybase.go</file> implements a custom model of the base component built-in resource, "mybase", and registers the new model and API helper functions with the Go Client SDK.
```
Comment (cheukt) [inline docs/extend/modular-resources/create/_index.md:177]: same edit as above. A module is also running as a server rather than a client, so Go SDK would be more appropriate
```suggestion
<file>mybase.go</file> implements "mybase", a custom model of the base component,  and registers the new model and API helper functions with the Go SDK.
```

File: docs/extend/modular-resources/create/_index.md [hunk 5]
```diff
@@ -493,15 +473,22 @@ For example:
...
 # or handle forwarding termination signals manually
-exec python3 -m <your-module-directory-name>.<main-program-filename-without-extension> $@
+exec $PYTHON src/main.py $@
 ```
 
 To make this shell script executable, run the following command in your terminal:
 
 ``` sh {id="terminal-prompt" class="command-line" data-prompt="$"}
-sudo chmod +x <FILEPATH>/<FILENAME>
+sudo chmod +x <file-path-to-your-main-program>/<your-main-program-filename>
```
Comment (cheukt) [inline docs/extend/modular-resources/create/_index.md:491]:
```suggestion
sudo chmod +x <file-path-to-entrypoint>
```
maybe? the current text could imply `main.py`, when we want `run.sh`

**Review by cheukt — CHANGES_REQUESTED** [review]
generally looks good, as long as we've ran the example end-to-end to test

File: docs/extend/modular-resources/create/_index.md (hunk 5 above)
Comment (sguequierre) [reply docs/extend/modular-resources/create/_index.md:491]: Oh yeah thank you!

File: docs/extend/modular-resources/create/_index.md (hunk 1 above)
Comment (sguequierre) [reply docs/extend/modular-resources/create/_index.md:66]: That's good with me, thank you for the suggestion!

File: docs/extend/modular-resources/create/_index.md (hunk 4 above)
Comment (sguequierre) [reply docs/extend/modular-resources/create/_index.md:177]: Cool, LGTM!

Comment (sguequierre) [reply docs/extend/modular-resources/create/_index.md:177]: done-- sorry for comment spam but i lose track of docs feedback sometimes if I don't comment when I finish or actually commit things 😵‍💫

File: docs/extend/modular-resources/create/_index.md (hunk 5 above)
Comment (sguequierre) [reply docs/extend/modular-resources/create/_index.md:491]: done

File: docs/extend/modular-resources/create/_index.md [hunk 6]
```diff
@@ -493,15 +473,22 @@ For example:
...
 # or handle forwarding termination signals manually
-exec python3 -m <your-module-directory-name>.<main-program-filename-without-extension> $@
+exec $PYTHON src/main.py $@
 ```
 
 To make this shell script executable, run the following command in your terminal:
 
 ``` sh {id="terminal-prompt" class="command-line" data-prompt="$"}
-sudo chmod +x <FILEPATH>/<FILENAME>
+sudo chmod +x <file-path-to-entrypoint>
```
Comment (sguequierre) [inline docs/extend/modular-resources/create/_index.md:491]:
```suggestion
sudo chmod +x <your-file-path-to>/run.sh
```

**Review by andf-viam — APPROVED** [review]
LGTM, just one comment about `venv`. Thanks!!

File: docs/extend/modular-resources/create/_index.md [hunk 7]
```diff
@@ -493,15 +473,22 @@ For example:
 #!/bin/sh
 cd `dirname $0`
 
+# Create a virtual environment to run our code
+VENV_NAME="venv"
```
[the remark this answers is not in the document]
Comment (cheukt) [reply docs/extend/modular-resources/create/_index.md:477]: isn't `venv` supplied with all pythons 3.3 and above? https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/#installing-virtualenv, since we only support >3.8, they should come with python if it is installed

File: docs/extend/modular-resources/create/_index.md [hunk 8]
```diff
@@ -63,24 +63,26 @@ The Go code for the custom model (<file>mybase.go</file>) and module entry point
 {{< tabs name="Sample SDK Code">}}
 {{% tab name="Python"%}}
 
-<file>my_base.py</file> implements a custom model of the base component built-in resource, "mybase".
+<file>my_base_client.py</file> implements a custom model of the base component built-in resource, "mybase".
 
 <details>
-  <summary>Click to view sample code from <file>my_base.py</file></summary>
+  <summary>Click to view sample code from <file>my_base_client.py</file></summary>
```
[the remark this answers is not in the document]
Comment (sguequierre) [reply docs/extend/modular-resources/create/_index.md:69]: Awesome thank you andrew, this would have caused some grief 🤣

File: docs/extend/modular-resources/create/_index.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (andf-viam) [reply docs/extend/modular-resources/create/_index.md:477]: You're right, thank you! I withdraw that comment!

File: docs/extend/modular-resources/create/_index.md (hunk 3 above)
Comment (sguequierre) [reply docs/extend/modular-resources/create/_index.md]: Done!

File: docs/extend/modular-resources/create/_index.md [hunk 9]
```diff
@@ -493,15 +473,22 @@ For example:
...
-exec python3 -m <your-module-directory-name>.<main-program-filename-without-extension> $@
+exec $PYTHON src/main.py $@
 ```
 
 To make this shell script executable, run the following command in your terminal:
 
 ``` sh {id="terminal-prompt" class="command-line" data-prompt="$"}
-sudo chmod +x <FILEPATH>/<FILENAME>
+sudo chmod +x <your-file-path-to>/<run.sh>
 ```
```
Comment (sguequierre) [inline docs/extend/modular-resources/create/_index.md:492]:
```suggestion
```

To make your main program executable, run the following command in your terminal:

``` sh {id="terminal-prompt" class="command-line" data-prompt="$"}
sudo chmod +x <FILEPATH>/<FILENAME>
sudo chmod +x <your-file-path-to>/<main.py>
```
```

**Review by cheukt — APPROVED** [review]
LGTM thanks for updating!
